Refactor XFEM geometric markers to be based on ElementUserObject

bwspenc opened the pull request:
closes #10555

This switches the GeometricCutUserObject and derived classes from being derived from a GeneralUserObject to being derived from an ElementUserObject. This allows for cuts to be based on information that is only available on the owning processor of an element being considered for cutting.

bwspenc committed:
Refactor XFEM geometric markers to be based on ElementUserObject closes #10555

bwspenc commented:
@jiangwen84 That problematic test started passing once I rebased this and removed another unrelated commit that I had in my history. It must have been something else in that other commit that broke it.

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/147218/) on c07dc4e wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/10557/site)

This comment will be updated on new commits.

bwspenc committed:
Other cleanup for XFEM geometric markers refactor closes #10555

Remove XFEMMarkerUserObject.md
Remove usage of 'geom_cut_user_objects' command from input files

permcody commented:
Closing due to inactivity (>30 days).
